### ISSUE TITLE
Monitoring v2 stack trace when not installed to Project: System by default in Rancher v2.5.x

### ISSUE THREAD

**Issue by oxr463** [rancher/rancher#32017]
**What kind of request is this (question/bug/enhancement/feature request):** Bug

**Steps to reproduce (least amount of steps as possible):**

1. Navigate to Apps & Marketplace in Cluster Explorer, and install Monitoring.
2. Navigate to Projects/Namespaces in Cluster Manager.

**Result:**

![image](https://user-images.githubusercontent.com/43783393/114221972-fce63e80-993b-11eb-8179-b229aeed44d3.png)

**Other details that may be helpful:**

```
E0331 20:45:07.406901       1 runtime.go:76] Observed a panic: runtime error: invalid memory address or nil pointer dereference 
goroutine 4518653 [running]: 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1.1(0xc006ee6180) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:108 +0x107 
panic(0x1861b80, 0x2cbc140) 
 /usr/local/go/src/runtime/panic.go:679 +0x1b2 
compress/gzip.(*Writer).Write(0xc00024fe40, 0xc006e34800, 0x71, 0x71, 0x30, 0x1926540, 0xc006decd01) 
 /usr/local/go/src/compress/gzip/gzip.go:168 +0x237 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.(*deferredResponseWriter).Write(0xc00054e7d0, 0xc006e34800, 0x71, 0x71, 0xc006e34800, 0x71, 0x71) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:182 +0x54e 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.SerializeObject(0x1ab94aa, 0x10, 0x7f4372acab78, 0xc006e52460, 0x7f4372aca500, 0xc00000e138, 0xc00001d800, 0xc8, 0x1d75180, 0xc00054e780) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:117 +0x389 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated(0x1d94fc0, 0xc0002d9ec0, 0x1d95200, 0x2cfcd78, 0x0, 0x0, 0x0, 0x0, 0x7f4372aca500, 0xc00000e138, ...) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:251 +0x555 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery.(*APIVersionHandler).ServeHTTP(0xc00004d500, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery/version.go:81 +0x18f 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery.(*APIVersionHandler).handle(...) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery/version.go:77 
github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful.(*Container).dispatch(0xc00003bc20, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful/container.go:288 +0xa4f 
github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful.(*Container).Dispatch(...) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful/container.go:199 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x1ac73ea, 0x1a, 0xc00003bc20, 0xc00024c690, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/handler.go:146 +0x4d3 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x512 
net/http.HandlerFunc.ServeHTTP(0xc000494280, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x5dc 
net/http.HandlerFunc.ServeHTTP(0xc0004493e0, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:50 +0x1fe6 
net/http.HandlerFunc.ServeHTTP(0xc0004942c0, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d700) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:131 +0xa8c 
net/http.HandlerFunc.ServeHTTP(0xc00011ef50, 0x7f4372aca500, 0xc00000e138, 0xc00001d700) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc006ee6180, 0xc00078a2a0, 0x1d9e200, 0xc00000e138, 0xc00001d700) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:113 +0xd0 
created by github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:99 +0x1cb 
 
goroutine 4518650 [running]: 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x17cd020, 0xc0076aa070) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0xc0004bbc90, 0x1, 0x1) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82 
panic(0x17cd020, 0xc0076aa070) 
 /usr/local/go/src/runtime/panic.go:679 +0x1b2 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP(0xc00078a2a0, 0x1d93180, 0xc0001a2150, 0xc00001d700) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:119 +0x486 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.WithWaitGroup.func1(0x1d93180, 0xc0001a2150, 0xc00001d600) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/waitgroup.go:47 +0x10f 
net/http.HandlerFunc.ServeHTTP(0xc000449410, 0x1d93180, 0xc0001a2150, 0xc00001d600) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithRequestInfo.func1(0x1d93180, 0xc0001a2150, 0xc00001d500) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/requestinfo.go:39 +0x274 
net/http.HandlerFunc.ServeHTTP(0xc0004494a0, 0x1d93180, 0xc0001a2150, 0xc00001d500) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/httplog.WithLogging.func1(0x1d87100, 0xc00000e118, 0xc00001d000) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:89 +0x2ca 
net/http.HandlerFunc.ServeHTTP(0xc00078a2c0, 0x1d87100, 0xc00000e118, 0xc00001d000) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.withPanicRecovery.func1(0x1d87100, 0xc00000e118, 0xc00001d000) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/wrap.go:51 +0x13e 
net/http.HandlerFunc.ServeHTTP(0xc00078a2e0, 0x1d87100, 0xc00000e118, 0xc00001d000) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server.(*APIServerHandler).ServeHTTP(0xc0004494d0, 0x1d87100, 0xc00000e118, 0xc00001d000) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/handler.go:189 +0x51 
net/http.serverHandler.ServeHTTP(0xc0008f8460, 0x1d87100, 0xc00000e118, 0xc00001d000) 
 /usr/local/go/src/net/http/server.go:2802 +0xa4 
net/http.initNPNRequest.ServeHTTP(0x1d9ddc0, 0xc0004ef140, 0xc0008b4380, 0xc0008f8460, 0x1d87100, 0xc00000e118, 0xc00001d000) 
 /usr/local/go/src/net/http/server.go:3366 +0x8d 
github.com/directxman12/k8s-prometheus-adapter/vendor/golang.org/x/net/http2.(*serverConn).runHandler(0xc0006ee000, 0xc00000e118, 0xc00001d000, 0xc006d0fee0) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/golang.org/x/net/http2/server.go:2149 +0x9f 
created by github.com/directxman12/k8s-prometheus-adapter/vendor/golang.org/x/net/http2.(*serverConn).processHeaders 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/golang.org/x/net/http2/server.go:1883 +0x4eb 
E0331 20:45:07.406948       1 wrap.go:39] apiserver panic'd on GET /apis/custom.metrics.k8s.io/v1beta1 
I0331 20:45:07.407017       1 log.go:172] http2: panic serving 169.253.0.0:49106: runtime error: invalid memory address or nil pointer dereference 
goroutine 4518653 [running]: 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1.1(0xc006ee6180) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:108 +0x107 
panic(0x1861b80, 0x2cbc140) 
 /usr/local/go/src/runtime/panic.go:679 +0x1b2 
compress/gzip.(*Writer).Write(0xc00024fe40, 0xc006e34800, 0x71, 0x71, 0x30, 0x1926540, 0xc006decd01) 
 /usr/local/go/src/compress/gzip/gzip.go:168 +0x237 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.(*deferredResponseWriter).Write(0xc00054e7d0, 0xc006e34800, 0x71, 0x71, 0xc006e34800, 0x71, 0x71) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:182 +0x54e 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.SerializeObject(0x1ab94aa, 0x10, 0x7f4372acab78, 0xc006e52460, 0x7f4372aca500, 0xc00000e138, 0xc00001d800, 0xc8, 0x1d75180, 0xc00054e780) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:117 +0x389 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated(0x1d94fc0, 0xc0002d9ec0, 0x1d95200, 0x2cfcd78, 0x0, 0x0, 0x0, 0x0, 0x7f4372aca500, 0xc00000e138, ...) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go:251 +0x555 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery.(*APIVersionHandler).ServeHTTP(0xc00004d500, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery/version.go:81 +0x18f 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery.(*APIVersionHandler).handle(...) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/discovery/version.go:77 
github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful.(*Container).dispatch(0xc00003bc20, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful/container.go:288 +0xa4f 
github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful.(*Container).Dispatch(...) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/github.com/emicklei/go-restful/container.go:199 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x1ac73ea, 0x1a, 0xc00003bc20, 0xc00024c690, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/handler.go:146 +0x4d3 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x512 
net/http.HandlerFunc.ServeHTTP(0xc000494280, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x5dc 
net/http.HandlerFunc.ServeHTTP(0xc0004493e0, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:50 +0x1fe6 
net/http.HandlerFunc.ServeHTTP(0xc0004942c0, 0x7f4372aca500, 0xc00000e138, 0xc00001d800) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0x7f4372aca500, 0xc00000e138, 0xc00001d700) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:131 +0xa8c 
net/http.HandlerFunc.ServeHTTP(0xc00011ef50, 0x7f4372aca500, 0xc00000e138, 0xc00001d700) 
 /usr/local/go/src/net/http/server.go:2007 +0x44 
github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc006ee6180, 0xc00078a2a0, 0x1d9e200, 0xc00000e138, 0xc00001d700) 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:113 +0xd0 
created by github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP 
 /go/src/github.com/directxman12/k8s-prometheus-adapter/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:99 +0x1cb
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.7
- Installation option (single install/HA): HA

gz#16066

**Comment by slickwarren** [comment]
on v2.5.8-rc18, I installed monitoring on a downstream cluster without a project in the latest version (v14.5.100) and the version released in 2.5.7 (v9.4.203) and was unable to reproduce this issue. 
I monitored the rancher logs throughout install of monitoring, and did not see a panic for either version.

**Comment by MKlimuszka** [comment]
QA could not reproduce and this was reported in a version that is no longer supported.
